// mdast-util-to-hast/lib/handlers/index.js
'use strict'

module.exports = {
  blockquote: require('mdast-util-to-hast/lib/handlers/blockquote'),
  break: require('mdast-util-to-hast/lib/handlers/break'),
  code: require('mdast-util-to-hast/lib/handlers/code'),
  delete: require('mdast-util-to-hast/lib/handlers/delete'),
  emphasis: require('mdast-util-to-hast/lib/handlers/emphasis'),
  footnoteReference: require('mdast-util-to-hast/lib/handlers/footnote-reference'),
  footnote: require('mdast-util-to-hast/lib/handlers/footnote'),
  heading: require('mdast-util-to-hast/lib/handlers/heading'),
  html: require('mdast-util-to-hast/lib/handlers/html'),
  imageReference: require('mdast-util-to-hast/lib/handlers/image-reference'),
  image: require('mdast-util-to-hast/lib/handlers/image'),
  inlineCode: require('mdast-util-to-hast/lib/handlers/inline-code'),
  linkReference: require('mdast-util-to-hast/lib/handlers/link-reference'),
  link: require('mdast-util-to-hast/lib/handlers/link'),
  listItem: require('mdast-util-to-hast/lib/handlers/list-item'),
  list: require('mdast-util-to-hast/lib/handlers/list'),
  paragraph: require('mdast-util-to-hast/lib/handlers/paragraph'),
  root: require('mdast-util-to-hast/lib/handlers/root'),
  strong: require('mdast-util-to-hast/lib/handlers/strong'),
  table: require('mdast-util-to-hast/lib/handlers/table'),
  text: require('mdast-util-to-hast/lib/handlers/text'),
  thematicBreak: require('mdast-util-to-hast/lib/handlers/thematic-break'),
  toml: ignore,
  yaml: ignore,
  definition: ignore,
  footnoteDefinition: ignore
}

// Return nothing for nodes that are ignored.
function ignore() {
  return null
}
