// @babel/types/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  react: true,
  assertNode: true,
  createTypeAnnotationBasedOnTypeof: true,
  createUnionTypeAnnotation: true,
  createFlowUnionType: true,
  createTSUnionType: true,
  cloneNode: true,
  clone: true,
  cloneDeep: true,
  cloneDeepWithoutLoc: true,
  cloneWithoutLoc: true,
  addComment: true,
  addComments: true,
  inheritInnerComments: true,
  inheritLeadingComments: true,
  inheritsComments: true,
  inheritTrailingComments: true,
  removeComments: true,
  ensureBlock: true,
  toBindingIdentifierName: true,
  toBlock: true,
  toComputedKey: true,
  toExpression: true,
  toIdentifier: true,
  toKeyAlias: true,
  toSequenceExpression: true,
  toStatement: true,
  valueToNode: true,
  appendToMemberExpression: true,
  inherits: true,
  prependToMemberExpression: true,
  removeProperties: true,
  removePropertiesDeep: true,
  removeTypeDuplicates: true,
  getBindingIdentifiers: true,
  getOuterBindingIdentifiers: true,
  traverse: true,
  traverseFast: true,
  shallowEqual: true,
  is: true,
  isBinding: true,
  isBlockScoped: true,
  isImmutable: true,
  isLet: true,
  isNode: true,
  isNodesEquivalent: true,
  isPlaceholderType: true,
  isReferenced: true,
  isScope: true,
  isSpecifierDefault: true,
  isType: true,
  isValidES3Identifier: true,
  isValidIdentifier: true,
  isVar: true,
  matchesPattern: true,
  validate: true,
  buildMatchMemberExpression: true
};
Object.defineProperty(exports, "assertNode", {
  enumerable: true,
  get: function () {
    return _assertNode.default;
  }
});
Object.defineProperty(exports, "createTypeAnnotationBasedOnTypeof", {
  enumerable: true,
  get: function () {
    return _createTypeAnnotationBasedOnTypeof.default;
  }
});
Object.defineProperty(exports, "createUnionTypeAnnotation", {
  enumerable: true,
  get: function () {
    return _createFlowUnionType.default;
  }
});
Object.defineProperty(exports, "createFlowUnionType", {
  enumerable: true,
  get: function () {
    return _createFlowUnionType.default;
  }
});
Object.defineProperty(exports, "createTSUnionType", {
  enumerable: true,
  get: function () {
    return _createTSUnionType.default;
  }
});
Object.defineProperty(exports, "cloneNode", {
  enumerable: true,
  get: function () {
    return _cloneNode.default;
  }
});
Object.defineProperty(exports, "clone", {
  enumerable: true,
  get: function () {
    return _clone.default;
  }
});
Object.defineProperty(exports, "cloneDeep", {
  enumerable: true,
  get: function () {
    return _cloneDeep.default;
  }
});
Object.defineProperty(exports, "cloneDeepWithoutLoc", {
  enumerable: true,
  get: function () {
    return _cloneDeepWithoutLoc.default;
  }
});
Object.defineProperty(exports, "cloneWithoutLoc", {
  enumerable: true,
  get: function () {
    return _cloneWithoutLoc.default;
  }
});
Object.defineProperty(exports, "addComment", {
  enumerable: true,
  get: function () {
    return _addComment.default;
  }
});
Object.defineProperty(exports, "addComments", {
  enumerable: true,
  get: function () {
    return _addComments.default;
  }
});
Object.defineProperty(exports, "inheritInnerComments", {
  enumerable: true,
  get: function () {
    return _inheritInnerComments.default;
  }
});
Object.defineProperty(exports, "inheritLeadingComments", {
  enumerable: true,
  get: function () {
    return _inheritLeadingComments.default;
  }
});
Object.defineProperty(exports, "inheritsComments", {
  enumerable: true,
  get: function () {
    return _inheritsComments.default;
  }
});
Object.defineProperty(exports, "inheritTrailingComments", {
  enumerable: true,
  get: function () {
    return _inheritTrailingComments.default;
  }
});
Object.defineProperty(exports, "removeComments", {
  enumerable: true,
  get: function () {
    return _removeComments.default;
  }
});
Object.defineProperty(exports, "ensureBlock", {
  enumerable: true,
  get: function () {
    return _ensureBlock.default;
  }
});
Object.defineProperty(exports, "toBindingIdentifierName", {
  enumerable: true,
  get: function () {
    return _toBindingIdentifierName.default;
  }
});
Object.defineProperty(exports, "toBlock", {
  enumerable: true,
  get: function () {
    return _toBlock.default;
  }
});
Object.defineProperty(exports, "toComputedKey", {
  enumerable: true,
  get: function () {
    return _toComputedKey.default;
  }
});
Object.defineProperty(exports, "toExpression", {
  enumerable: true,
  get: function () {
    return _toExpression.default;
  }
});
Object.defineProperty(exports, "toIdentifier", {
  enumerable: true,
  get: function () {
    return _toIdentifier.default;
  }
});
Object.defineProperty(exports, "toKeyAlias", {
  enumerable: true,
  get: function () {
    return _toKeyAlias.default;
  }
});
Object.defineProperty(exports, "toSequenceExpression", {
  enumerable: true,
  get: function () {
    return _toSequenceExpression.default;
  }
});
Object.defineProperty(exports, "toStatement", {
  enumerable: true,
  get: function () {
    return _toStatement.default;
  }
});
Object.defineProperty(exports, "valueToNode", {
  enumerable: true,
  get: function () {
    return _valueToNode.default;
  }
});
Object.defineProperty(exports, "appendToMemberExpression", {
  enumerable: true,
  get: function () {
    return _appendToMemberExpression.default;
  }
});
Object.defineProperty(exports, "inherits", {
  enumerable: true,
  get: function () {
    return _inherits.default;
  }
});
Object.defineProperty(exports, "prependToMemberExpression", {
  enumerable: true,
  get: function () {
    return _prependToMemberExpression.default;
  }
});
Object.defineProperty(exports, "removeProperties", {
  enumerable: true,
  get: function () {
    return _removeProperties.default;
  }
});
Object.defineProperty(exports, "removePropertiesDeep", {
  enumerable: true,
  get: function () {
    return _removePropertiesDeep.default;
  }
});
Object.defineProperty(exports, "removeTypeDuplicates", {
  enumerable: true,
  get: function () {
    return _removeTypeDuplicates.default;
  }
});
Object.defineProperty(exports, "getBindingIdentifiers", {
  enumerable: true,
  get: function () {
    return _getBindingIdentifiers.default;
  }
});
Object.defineProperty(exports, "getOuterBindingIdentifiers", {
  enumerable: true,
  get: function () {
    return _getOuterBindingIdentifiers.default;
  }
});
Object.defineProperty(exports, "traverse", {
  enumerable: true,
  get: function () {
    return _traverse.default;
  }
});
Object.defineProperty(exports, "traverseFast", {
  enumerable: true,
  get: function () {
    return _traverseFast.default;
  }
});
Object.defineProperty(exports, "shallowEqual", {
  enumerable: true,
  get: function () {
    return _shallowEqual.default;
  }
});
Object.defineProperty(exports, "is", {
  enumerable: true,
  get: function () {
    return _is.default;
  }
});
Object.defineProperty(exports, "isBinding", {
  enumerable: true,
  get: function () {
    return _isBinding.default;
  }
});
Object.defineProperty(exports, "isBlockScoped", {
  enumerable: true,
  get: function () {
    return _isBlockScoped.default;
  }
});
Object.defineProperty(exports, "isImmutable", {
  enumerable: true,
  get: function () {
    return _isImmutable.default;
  }
});
Object.defineProperty(exports, "isLet", {
  enumerable: true,
  get: function () {
    return _isLet.default;
  }
});
Object.defineProperty(exports, "isNode", {
  enumerable: true,
  get: function () {
    return _isNode.default;
  }
});
Object.defineProperty(exports, "isNodesEquivalent", {
  enumerable: true,
  get: function () {
    return _isNodesEquivalent.default;
  }
});
Object.defineProperty(exports, "isPlaceholderType", {
  enumerable: true,
  get: function () {
    return _isPlaceholderType.default;
  }
});
Object.defineProperty(exports, "isReferenced", {
  enumerable: true,
  get: function () {
    return _isReferenced.default;
  }
});
Object.defineProperty(exports, "isScope", {
  enumerable: true,
  get: function () {
    return _isScope.default;
  }
});
Object.defineProperty(exports, "isSpecifierDefault", {
  enumerable: true,
  get: function () {
    return _isSpecifierDefault.default;
  }
});
Object.defineProperty(exports, "isType", {
  enumerable: true,
  get: function () {
    return _isType.default;
  }
});
Object.defineProperty(exports, "isValidES3Identifier", {
  enumerable: true,
  get: function () {
    return _isValidES3Identifier.default;
  }
});
Object.defineProperty(exports, "isValidIdentifier", {
  enumerable: true,
  get: function () {
    return _isValidIdentifier.default;
  }
});
Object.defineProperty(exports, "isVar", {
  enumerable: true,
  get: function () {
    return _isVar.default;
  }
});
Object.defineProperty(exports, "matchesPattern", {
  enumerable: true,
  get: function () {
    return _matchesPattern.default;
  }
});
Object.defineProperty(exports, "validate", {
  enumerable: true,
  get: function () {
    return _validate.default;
  }
});
Object.defineProperty(exports, "buildMatchMemberExpression", {
  enumerable: true,
  get: function () {
    return _buildMatchMemberExpression.default;
  }
});
exports.react = void 0;

var _isReactComponent = _interopRequireDefault(require("@babel/types/lib/validators/react/isReactComponent"));

var _isCompatTag = _interopRequireDefault(require("@babel/types/lib/validators/react/isCompatTag"));

var _buildChildren = _interopRequireDefault(require("@babel/types/lib/builders/react/buildChildren"));

var _assertNode = _interopRequireDefault(require("@babel/types/lib/asserts/assertNode"));

var _generated = require("@babel/types/lib/asserts/generated");

Object.keys(_generated).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _generated[key];
    }
  });
});

var _createTypeAnnotationBasedOnTypeof = _interopRequireDefault(require("@babel/types/lib/builders/flow/createTypeAnnotationBasedOnTypeof"));

var _createFlowUnionType = _interopRequireDefault(require("@babel/types/lib/builders/flow/createFlowUnionType"));

var _createTSUnionType = _interopRequireDefault(require("@babel/types/lib/builders/typescript/createTSUnionType"));

var _generated2 = require("@babel/types/lib/builders/generated");

Object.keys(_generated2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _generated2[key];
    }
  });
});

var _cloneNode = _interopRequireDefault(require("@babel/types/lib/clone/cloneNode"));

var _clone = _interopRequireDefault(require("@babel/types/lib/clone/clone"));

var _cloneDeep = _interopRequireDefault(require("@babel/types/lib/clone/cloneDeep"));

var _cloneDeepWithoutLoc = _interopRequireDefault(require("@babel/types/lib/clone/cloneDeepWithoutLoc"));

var _cloneWithoutLoc = _interopRequireDefault(require("@babel/types/lib/clone/cloneWithoutLoc"));

var _addComment = _interopRequireDefault(require("@babel/types/lib/comments/addComment"));

var _addComments = _interopRequireDefault(require("@babel/types/lib/comments/addComments"));

var _inheritInnerComments = _interopRequireDefault(require("@babel/types/lib/comments/inheritInnerComments"));

var _inheritLeadingComments = _interopRequireDefault(require("@babel/types/lib/comments/inheritLeadingComments"));

var _inheritsComments = _interopRequireDefault(require("@babel/types/lib/comments/inheritsComments"));

var _inheritTrailingComments = _interopRequireDefault(require("@babel/types/lib/comments/inheritTrailingComments"));

var _removeComments = _interopRequireDefault(require("@babel/types/lib/comments/removeComments"));

var _generated3 = require("@babel/types/lib/constants/generated");

Object.keys(_generated3).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _generated3[key];
    }
  });
});

var _constants = require("@babel/types/lib/constants");

Object.keys(_constants).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _constants[key];
    }
  });
});

var _ensureBlock = _interopRequireDefault(require("@babel/types/lib/converters/ensureBlock"));

var _toBindingIdentifierName = _interopRequireDefault(require("@babel/types/lib/converters/toBindingIdentifierName"));

var _toBlock = _interopRequireDefault(require("@babel/types/lib/converters/toBlock"));

var _toComputedKey = _interopRequireDefault(require("@babel/types/lib/converters/toComputedKey"));

var _toExpression = _interopRequireDefault(require("@babel/types/lib/converters/toExpression"));

var _toIdentifier = _interopRequireDefault(require("@babel/types/lib/converters/toIdentifier"));

var _toKeyAlias = _interopRequireDefault(require("@babel/types/lib/converters/toKeyAlias"));

var _toSequenceExpression = _interopRequireDefault(require("@babel/types/lib/converters/toSequenceExpression"));

var _toStatement = _interopRequireDefault(require("@babel/types/lib/converters/toStatement"));

var _valueToNode = _interopRequireDefault(require("@babel/types/lib/converters/valueToNode"));

var _definitions = require("@babel/types/lib/definitions");

Object.keys(_definitions).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _definitions[key];
    }
  });
});

var _appendToMemberExpression = _interopRequireDefault(require("@babel/types/lib/modifications/appendToMemberExpression"));

var _inherits = _interopRequireDefault(require("@babel/types/lib/modifications/inherits"));

var _prependToMemberExpression = _interopRequireDefault(require("@babel/types/lib/modifications/prependToMemberExpression"));

var _removeProperties = _interopRequireDefault(require("@babel/types/lib/modifications/removeProperties"));

var _removePropertiesDeep = _interopRequireDefault(require("@babel/types/lib/modifications/removePropertiesDeep"));

var _removeTypeDuplicates = _interopRequireDefault(require("@babel/types/lib/modifications/flow/removeTypeDuplicates"));

var _getBindingIdentifiers = _interopRequireDefault(require("@babel/types/lib/retrievers/getBindingIdentifiers"));

var _getOuterBindingIdentifiers = _interopRequireDefault(require("@babel/types/lib/retrievers/getOuterBindingIdentifiers"));

var _traverse = _interopRequireDefault(require("@babel/types/lib/traverse/traverse"));

var _traverseFast = _interopRequireDefault(require("@babel/types/lib/traverse/traverseFast"));

var _shallowEqual = _interopRequireDefault(require("@babel/types/lib/utils/shallowEqual"));

var _is = _interopRequireDefault(require("@babel/types/lib/validators/is"));

var _isBinding = _interopRequireDefault(require("@babel/types/lib/validators/isBinding"));

var _isBlockScoped = _interopRequireDefault(require("@babel/types/lib/validators/isBlockScoped"));

var _isImmutable = _interopRequireDefault(require("@babel/types/lib/validators/isImmutable"));

var _isLet = _interopRequireDefault(require("@babel/types/lib/validators/isLet"));

var _isNode = _interopRequireDefault(require("@babel/types/lib/validators/isNode"));

var _isNodesEquivalent = _interopRequireDefault(require("@babel/types/lib/validators/isNodesEquivalent"));

var _isPlaceholderType = _interopRequireDefault(require("@babel/types/lib/validators/isPlaceholderType"));

var _isReferenced = _interopRequireDefault(require("@babel/types/lib/validators/isReferenced"));

var _isScope = _interopRequireDefault(require("@babel/types/lib/validators/isScope"));

var _isSpecifierDefault = _interopRequireDefault(require("@babel/types/lib/validators/isSpecifierDefault"));

var _isType = _interopRequireDefault(require("@babel/types/lib/validators/isType"));

var _isValidES3Identifier = _interopRequireDefault(require("@babel/types/lib/validators/isValidES3Identifier"));

var _isValidIdentifier = _interopRequireDefault(require("@babel/types/lib/validators/isValidIdentifier"));

var _isVar = _interopRequireDefault(require("@babel/types/lib/validators/isVar"));

var _matchesPattern = _interopRequireDefault(require("@babel/types/lib/validators/matchesPattern"));

var _validate = _interopRequireDefault(require("@babel/types/lib/validators/validate"));

var _buildMatchMemberExpression = _interopRequireDefault(require("@babel/types/lib/validators/buildMatchMemberExpression"));

var _generated4 = require("@babel/types/lib/validators/generated");

Object.keys(_generated4).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _generated4[key];
    }
  });
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const react = {
  isReactComponent: _isReactComponent.default,
  isCompatTag: _isCompatTag.default,
  buildChildren: _buildChildren.default
};
exports.react = react;
