// remark-parse/lib/parser.js
'use strict'

var xtend = require('xtend')
var toggle = require('state-toggle')
var vfileLocation = require('vfile-location')
var unescape = require('remark-parse/lib/unescape')
var decode = require('remark-parse/lib/decode')
var tokenizer = require('remark-parse/lib/tokenizer')

module.exports = Parser

function Parser(doc, file) {
  this.file = file
  this.offset = {}
  this.options = xtend(this.options)
  this.setOptions({})

  this.inList = false
  this.inBlock = false
  this.inLink = false
  this.atStart = true

  this.toOffset = vfileLocation(file).toOffset
  this.unescape = unescape(this, 'escape')
  this.decode = decode(this)
}

var proto = Parser.prototype

// Expose core.
proto.setOptions = require('remark-parse/lib/set-options')
proto.parse = require('remark-parse/lib/parse')

// Expose `defaults`.
proto.options = require('remark-parse/lib/defaults')

// Enter and exit helpers.
proto.exitStart = toggle('atStart', true)
proto.enterList = toggle('inList', false)
proto.enterLink = toggle('inLink', false)
proto.enterBlock = toggle('inBlock', false)

// Nodes that can interupt a paragraph:
//
// ```markdown
// A paragraph, followed by a thematic break.
// ___
// ```
//
// In the above example, the thematic break “interupts” the paragraph.
proto.interruptParagraph = [
  ['thematicBreak'],
  ['list'],
  ['atxHeading'],
  ['fencedCode'],
  ['blockquote'],
  ['html'],
  ['setextHeading', {commonmark: false}],
  ['definition', {commonmark: false}]
]

// Nodes that can interupt a list:
//
// ```markdown
// - One
// ___
// ```
//
// In the above example, the thematic break “interupts” the list.
proto.interruptList = [
  ['atxHeading', {pedantic: false}],
  ['fencedCode', {pedantic: false}],
  ['thematicBreak', {pedantic: false}],
  ['definition', {commonmark: false}]
]

// Nodes that can interupt a blockquote:
//
// ```markdown
// > A paragraph.
// ___
// ```
//
// In the above example, the thematic break “interupts” the blockquote.
proto.interruptBlockquote = [
  ['indentedCode', {commonmark: true}],
  ['fencedCode', {commonmark: true}],
  ['atxHeading', {commonmark: true}],
  ['setextHeading', {commonmark: true}],
  ['thematicBreak', {commonmark: true}],
  ['html', {commonmark: true}],
  ['list', {commonmark: true}],
  ['definition', {commonmark: false}]
]

// Handlers.
proto.blockTokenizers = {
  blankLine: require('remark-parse/lib/tokenize/blank-line'),
  indentedCode: require('remark-parse/lib/tokenize/code-indented'),
  fencedCode: require('remark-parse/lib/tokenize/code-fenced'),
  blockquote: require('remark-parse/lib/tokenize/blockquote'),
  atxHeading: require('remark-parse/lib/tokenize/heading-atx'),
  thematicBreak: require('remark-parse/lib/tokenize/thematic-break'),
  list: require('remark-parse/lib/tokenize/list'),
  setextHeading: require('remark-parse/lib/tokenize/heading-setext'),
  html: require('remark-parse/lib/tokenize/html-block'),
  definition: require('remark-parse/lib/tokenize/definition'),
  table: require('remark-parse/lib/tokenize/table'),
  paragraph: require('remark-parse/lib/tokenize/paragraph')
}

proto.inlineTokenizers = {
  escape: require('remark-parse/lib/tokenize/escape'),
  autoLink: require('remark-parse/lib/tokenize/auto-link'),
  url: require('remark-parse/lib/tokenize/url'),
  email: require('remark-parse/lib/tokenize/email'),
  html: require('remark-parse/lib/tokenize/html-inline'),
  link: require('remark-parse/lib/tokenize/link'),
  reference: require('remark-parse/lib/tokenize/reference'),
  strong: require('remark-parse/lib/tokenize/strong'),
  emphasis: require('remark-parse/lib/tokenize/emphasis'),
  deletion: require('remark-parse/lib/tokenize/delete'),
  code: require('remark-parse/lib/tokenize/code-inline'),
  break: require('remark-parse/lib/tokenize/break'),
  text: require('remark-parse/lib/tokenize/text')
}

// Expose precedence.
proto.blockMethods = keys(proto.blockTokenizers)
proto.inlineMethods = keys(proto.inlineTokenizers)

// Tokenizers.
proto.tokenizeBlock = tokenizer('block')
proto.tokenizeInline = tokenizer('inline')
proto.tokenizeFactory = tokenizer

// Get all keys in `value`.
function keys(value) {
  var result = []
  var key

  for (key in value) {
    result.push(key)
  }

  return result
}
